### DISCUSSION TITLE
yaml expression syntax

### DISCUSSION THREAD

**Discussion by dirkhh** [community/community#25557] [Actions]
<p>The documentation of expressions in yaml is… not leaving me with the feeling that I know what I’m doing.</p>
<p>Here’s what I want:</p>
<pre><code>jobs:
  build-container-stage1:
    runs-on: ubuntu-latest
    env:
      VERSION: ${{ '0.2' }}
      NAME: ${{ 'subsurface/mxe-build-container:$VERSION.stage1' }}
</code></pre>
<p>I have tried at least a dozen ways to do this - with join() expressions, with stage.output constructrs… nothing seems to work.</p>
<p>What would be really useful is a complex, well documented YAML file that exercises many of the things that people might want to use…</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Thank you for your suggestion.</p>
<p>If you want to use VERSION environment variable in NAME , I would suggest you use ${{env.VERSION}} instead of $VERSION in step.env. Please check my example:</p>
<p>jobs:</p>
<p>  job1:</p>
<p>    runs-on: ubuntu-latest</p>
<p>    env:</p>
<p>      VERSION: ${{ ‘0.2’ }}</p>
<p>    </p>
<p>    steps:</p>
<p>    - uses: actions/checkout@v1</p>
<p>    - run: echo ‘version’  $VERSION  ‘name’ $NAME</p>
<p>       env:</p>
<p>          NAME: subsurface/mxe-build-container:${{env.VERSION}}.stage1</p>
<p>There is a similar issue for how to use env context：<a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.community%2Ft5%2FGitHub-Actions%2FHow-to-use-env-context%2Ftd-p%2F38951&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7C6e1b0ea261f54f43834c08d7717942fb%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637102637937397070&amp;sdata=9HoizKVOtp%2BngIJqXk4azDbdXMxePprUGbw04ucedQs%3D&amp;reserved=0" rel="nofollow noopener">https://github.community/t5/GitHub-Actions/How-to-use-env-context/td-p/38951</a></p>
<p>I have submitted a suggestion to appropriate engineering team  . Ask them to make it clear and add some examples for using env context to <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions" rel="nofollow noopener">this</a> document.  </p>
